src/main/java/com/elearningbackend/service/QuestionService.java

```diff
@@ -0,0 +1,232 @@
+        /**
+         * Check this question has child
+         * if result is true
+         * => change children's question code by new parent's question code
+         * Purpose: don't set parent's question code for each children question
+         */
+        if(hasChildQuestion(questionDtos,questionBankDto.getQuestionCode())){
+            changeQuestionParentCode(questionDtos,oldQuestionCode
+                    , questionDto.getQuestionBankDto().getQuestionCode());
+        }
+    }
+
+    private boolean hasCorrectAnswer(List<AnswerDto> answerDtos){
+        return answerDtos.stream().anyMatch(e -> e.getSystemResultDto().getSystemResultIsCorrect()==Constants.ANSWER_CORRECT);
+    }
+
+    private boolean hasChildQuestion(List<QuestionDto> questionDtos, String questiongCode){
+        return questionDtos.stream().anyMatch(e -> e.getQuestionBankDto().getQuestionParentCode()!=null && e.getQuestionBankDto().getQuestionParentCode().equals(questiongCode));
+    }
+
+    private void changeQuestionParentCode(List<QuestionDto> questionDtos, String oldQuestionCode, String newQuestionCode){
+        questionDtos.stream().map(e -> {
+            if(e.getQuestionBankDto().getQuestionCode().equals(oldQuestionCode)){
+                e.getQuestionBankDto().setQuestionCode(newQuestionCode);
+            }
+            return e;
+        });
+    }
+
+    private List<AnswerDto> addAnswerDtoAndSystemResult(List<SystemResultDto> systemResultDtos,int isCorrect,int length){
+        List<AnswerDto> answerDtos= new ArrayList<>();
+        AnswerDto answerDto = null;
+        for (int i = 0, size = systemResultDtos.size(); i < size; i++) {
+            if(length > 0 && i >length)
+                break;
+            if(isCorrect==Constants.ANSWER_INCORRECT && systemResultDtos.get(i).getSystemResultIsCorrect() != Constants.ANSWER_INCORRECT)
+                continue;
+            if(isCorrect==Constants.ANSWER_CORRECT && systemResultDtos.get(i).getSystemResultIsCorrect() != Constants.ANSWER_CORRECT)
+                continue;
+            answerDto = new AnswerDto();
+            answerDto.setAnswerBankDto(mapper.map(systemResultDtos.get(i).getAnswerBank(),AnswerBankDto.class));
+            ignoreSystemResultDto(systemResultDtos.get(i));
+            answerDto.setSystemResultDto(systemResultDtos.get(i));
+            answerDtos.add(answerDto);
+        }
+        return answerDtos;
+    }
+
+    private void ignoreSystemResultDto(SystemResultDto systemResultDto){
+        systemResultDto.setQuestionBank(null);
+        systemResultDto.setAnswerBank(null);
+    }
+
+    private void editSystemResullt(QuestionDto questionDto) throws ElearningMapException {
+        Map<Integer,ElearningException> errorMap = new   HashMap<>();
+        if(questionDto.getQuestionBankDto().getQuestionType() != Constants.Q_TYPE_PARAGRAPH){
+            for (Integer i = 0, length = questionDto.getAnswerDtos().size(); i < length; i++) {
```

Comment:
dùng kiểu nguyên thủy cho đồng nhất nào

Reply:
Map hình như khong cho dùng kiểu nguyên thủy int

Reply:
ý a là chỗ này đó for (Integer i = 0, length = questionDto.getAnswerDtos()...
Java nó tự autoboxing rồi nên k cần Integer

Reply:
cái i cần ở dưới để thêm key cho map

Reply:
vẫn thêm đc. a bảo nó autoboxing mà